instaclustr/resource_gcp_vpc_peering.go

```diff
@@ -0,0 +1,166 @@
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"vpc_peering_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"cdc_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"peer_vpc_network_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"peer_project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"peer_subnet": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"peer_subnets"},
+			},
+
+			"peer_subnets": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:      true,
+				ConflictsWith: []string{"peer_subnet"},
+			},
+		},
+	}
+}
+
+func GCPresourceVpcPeeringCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Config).Client
+
+	cdcID, err := VpcPeeringCreate(d, meta)
+	if err != nil {
+		return fmt.Errorf("[Error] Error creating VPC peering request: %s", err)
+	}
+	var createData CreateGCPVPCPeeringRequest
+	createData, err = GCPcreateVpcPeeringRequest(d)
+	if err != nil {
+		return fmt.Errorf("[Error] Error creating GCP VPC peering request: %s", err)
+	}
+
+	var jsonStr []byte
+	jsonStr, err = json.Marshal(createData)
+	if err != nil {
+		return fmt.Errorf("[Error] Error creating GCP VPC peering request: %s", err)
+	}
+	var id string
+	id, err = client.CreateVpcPeering(cdcID, jsonStr)
+	if err != nil {
+		return fmt.Errorf("[Error] Error in creating the cluster: %s", err)
+	}
+	d.SetId(id)
+	d.Set("vpc_peering_id", id)
+	d.Set("cdc_id", cdcID)
+	var vpcPeering *GCPVPCPeering
+	vpcPeering, err = client.GCPReadVpcPeering(cdcID, id)
+	if err != nil {
+		return fmt.Errorf("[Error] Error in reading GCP VPC peering connection: %s", err)
+	}
+
+	d.Set("peer_project_id", vpcPeering.PeerProjectID)
+
+	log.Printf("[INFO] VPC peering request %s has been created.", id)
+	return nil
+}
+
+func GCPresourceVpcPeeringRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Config).Client
+	cluster, err := client.ReadCluster(d.Get("cluster_id").(string))
+	if err != nil {
+		return fmt.Errorf("[Error] Error retrieving cluster info: %s", err)
+	}
+	cdcID := cluster.DataCentres[0].ID
+	vpcPeeringID := d.Get("vpc_peering_id").(string)
+
+	log.Printf("[INFO] Reading the status of VPC peering connection %s.", vpcPeeringID)
+	vpcPeering, err := client.GCPReadVpcPeering(cdcID, vpcPeeringID)
+	if err != nil {
+		return fmt.Errorf("[Error] Error reading VPC peering connection: %s", err)
+	}
+	err = GCPresourceVpcPeeringReadHelper(d, vpcPeering)
+	if err != nil {
+		return fmt.Errorf("[Error] Error reading VPC peering connection: %s", err)
+	}
+	return nil
+
+}
+
+func GCPresourceVpcPeeringReadHelper(d *schema.ResourceData, vpcPeering *GCPVPCPeering) error {
```

Comment:
**[Maintainability,Desired]** The naming of this func doesn't really tell me what it does. I can see from the body that it maps a GCPVPCPeering resource instance to a schema, but I shouldn't need to read into the body of it to know that. Might be better to name it sometime like `MapGCPVPCPeeringToResource`

Reply:
Done